com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetLayout.java

```diff
@@ -42,50 +42,168 @@ public String toString() {
         }
     }
 
-    public static class Rect {
-        public String id;
-        public int index; // for easier keeping the original order
-        public int x, y, width, height;
-        public int page;
-        public boolean rotated;
+    public static class Pointi
+    {
+        public int x;
+        public int y;
+        public Pointi() {};
+        public Pointi(int x, int y) {
+            this.x = x;
+            this.y = y;
+        };
+    }
 
-        public Rect(String id, int index, int x, int y, int width, int height) {
-            this.id = id;
-            this.index = index;
+    public static class Sizei
+    {
+        public int width;
+        public int height;
+        public Sizei(int width, int height) {
+            this.width = width;
+            this.height = height;
+        };
+    }
+
+    public static class Rectanglei
+    {
+        public int x;
+        public int y;
+        public int width;
+        public int height;
+
+        public Rectanglei(int x, int y, int width, int height) {
             this.x = x;
             this.y = y;
             this.width = width;
             this.height = height;
-            this.rotated = false;
+        };
+
+        public Point getCenter() {
+            return new Point(x + width * 0.5f, y + height * 0.5f);
         }
 
-        public Rect(String id, int index, int width, int height) {
+        public void addBorder(int border) {
+            x -= border;
+            y -= border;
+            width += border * 2;
+            height += border * 2;
+        }
+    }
+
+    // TODO: Rename this class?
+    public static class Rect {
```

Comment:
This name is not really representative anymore, and I should rename it. Perhaps as an MVP2.